sql/src/test/scala/org/datasyslab/geosparksql/functionTestScala.scala

```diff
@@ -285,5 +285,30 @@ class functionTestScala extends TestBaseScala {
       var test = sparkSession.sql("SELECT ST_GeometryType(ST_GeomFromText('LINESTRING(77.29 29.07,77.42 29.26,77.27 29.31,77.29 29.07)'))")
       assert(test.take(1)(0).get(0).asInstanceOf[String].toUpperCase() == "ST_LINESTRING")
     }
+
+    it("Passed ST_NumGeometries"){
+      import sparkSession.implicits._
+
+      // Test data
+      val m1 = "LINESTRING (-29 -27, -30 -29.7, -45 -33)"
+      val m2 = "MULTILINESTRING ((-29 -27, -30 -29.7, -36 -31, -45 -33), (-45.2 -33.2, -46 -32))"
+      val p1 = "POLYGON ((8 25, 28 22, 15 11, 33 3, 56 30, 47 44, 35 36, 43 19, 24 39, 8 25))"
+      val p2 = "MULTIPOLYGON(((0 0, 3 0, 3 3, 0 3, 0 0)), ((3 0, 6 0, 6 3, 3 3, 3 0)))"
+      val g1 = "GEOMETRYCOLLECTION(MULTIPOINT(-2 3 , -2 2), LINESTRING(5 5 ,10 10), POLYGON((-7 4.2,-7.1 5,-7.1 4.3,-7 4.2)))"
+
+
+      val testData = Seq((m1), (m2), (p1), (p2), (g1)).toDF("Geometry")
+
+      // Call transformation
+      var testDF = testData.selectExpr("Geometry", "ST_NumGeometries(ST_GeomFromText(Geometry)) as geom")
+      testDF.show(false)
+
+      val result = testDF.take(5)
```

Comment:
You used spark implicits, with scala matchers assert can be used like 
```scala
testData.selectExpr("ST_NumGeometries(ST_GeomFromText(Geometry)) as geom").as[Int].collect().toList should contain theSameElementsAs List(1, 2, 1, 2, 3)
```
you have example here https://github.com/Imbruced/GeoSpark/blob/additional-sql-functions/sql/src/test/scala/org/datasyslab/geosparksql/functionTestScala.scala#L683

Reply:
Thanks for your comments, @Imbruced, I like the GivenWhenThen approach, so I have follow the same approach here than in the link that you sent me.

I'll do the same in the other PR.